src/components/sortableGridView/index.tsx

```diff
@@ -0,0 +1,93 @@
+import React, {useMemo} from 'react';
+import SortableGridItemAnimationWrapper from './SortableGridItemAnimationWrapper';
+import {DEFAULT_NO_OF_COLUMNS, getItemSize, useSortableGridConfig, ItemsOrder, WINDOW_WIDTH} from './config';
+import Animated, {useAnimatedRef, useAnimatedScrollHandler, useSharedValue} from 'react-native-reanimated';
+import View from '../view';
+import {GestureHandlerRootView} from 'react-native-gesture-handler';
+
+
+export interface SortableGridItemProps {
+    /**
+     * unique identifier for a sortable grid item
+     */
+    id: string;
+}
+export interface SortableGridViewProps {
+    /**
+     * The list of itemProps to be rendered by renderItem
+     */
+    items: SortableGridItemProps[];
+    /** 
+    * Render method for sortable grid items,
+    * these components will be rendered inside the assigned space calculated by the grid
+    */
+    renderItem: (item: SortableGridItemProps) => React.ReactElement;
+    /**
+     * Number of items to show in a row
+     */
+    numOfColumns?: number;
+    /**
+     * Spacing between each item
+     */
+    itemSpacing?: number;
+    /**
+     * Pass the desired grid view width (default is screen wide)
+     */
+    viewWidth?: number;
+}
+
+const SortableGridView: React.FC<SortableGridViewProps> = (props) => {
+  const {items, renderItem, itemSpacing, numOfColumns = DEFAULT_NO_OF_COLUMNS, viewWidth = WINDOW_WIDTH} = props;
+  const scrollViewRef = useAnimatedRef<Animated.ScrollView>();
+  const scrollY = useSharedValue(0);
+  const itemSize = useMemo(() => getItemSize(numOfColumns, viewWidth), [viewWidth, numOfColumns]);
+  const {getPositionByOrder, getOrderByPosition} = useSortableGridConfig(itemSize, numOfColumns);
+  const itemsOrder = useSharedValue<ItemsOrder>(Object.assign({},
+    ...items.map((item, index) => ({[`${item.id} - ${index}`]: index}))));
+
+  const onScroll = useAnimatedScrollHandler({
+    onScroll: ({contentOffset}) => {
+      scrollY.value = contentOffset.y;
+    }
+  });
+
+  const renderGridItem = (item: SortableGridItemProps, index: number) => {
+    return (
+      <SortableGridItemAnimationWrapper
+        key={`${item.id} - ${index}`}
+        id={`${item.id} - ${index}`}
+        itemSize={itemSize}
+        itemsOrder={itemsOrder}
+        numOfColumns={numOfColumns}
+        scrollViewRef={scrollViewRef}
+        scrollY={scrollY}
+        getPositionByOrder={getPositionByOrder}
+        getOrderByPosition={getOrderByPosition}
+        itemSpacing={itemSpacing}
+      >
+        <View flex key={`${item.id} - ${index}`}>
+          {renderItem(item)}
+        </View>
+      </SortableGridItemAnimationWrapper>
+    );
+  };
+    
+  return (
+    <GestureHandlerRootView>
+      <Animated.ScrollView
```

Comment:
What's the need for ScrollView? (Also why is It Animated.ScrollView?)
Shouldn't the user decide whether to wrap it with ScrollView outside of the component?